sentry-core/src/main/java/io/sentry/core/SentryOptions.java

```diff
@@ -128,13 +128,13 @@
    * packages. Modules considered not to be part of the app will be hidden from stack traces by
    * default.
    */
-  private final @NotNull List<String> inAppExcludes = new ArrayList<>();
+  private final @NotNull List<String> inAppExcludes = new CopyOnWriteArrayList<>();
 
   /**
    * A list of string prefixes of module names that belong to the app. This option takes precedence
    * over inAppExcludes.
    */
-  private final @NotNull List<String> inAppIncludes = new ArrayList<>();
+  private final @NotNull List<String> inAppIncludes = new CopyOnWriteArrayList<>();
```

Comment:
m: We're not really making this class thread-safe just for changing this. This class wasn't designed to be access in multiple threads, and that's OK. The documented way is to create an instance, configure it, then call `Sentry.init` and not spawn a new thread to call `Sentry.init` while you mutate the options in concurrently.

If we decide to make this class thread-safe we'd need to make a pass across and make more changes.

So IMO it's not worth changing this type to `CopyOnWriteArrayList` which will allocate a new instance for each `add`.

Reply:
I get your point and agree, but we also don't offer another way of disabling all the default integrations but iterating and removing them.
this change is not meant to make SentryOptions fully thread-safe, but to avoid issues that we already know, iterating a non-thread-safe list and removing it immediately, which might be the 1st try of most developers, either because of not knowing how list work or just guessing its a thread-safe list.
IMO it's worth making this thread-safe list but also fine if not, I Just think this might become a recurrent rant.
cc @philipphofmann ? opinions?

Reply:
> way of disabling all the default integrations but iterating and removing them.
I see, they are mutating the iterator instead of creating a new list and setting it back.

It does make sense then to make this change since it's the most obvious way to change things here.

I'm unaware of a legitimate use case of "removing all default integrations". We could though have this list "add only" and control what gets activated though `SentryOptions`. We're sort of doing that already, except we allow Integrations to be removed now even if SentryOptions had that feature turned on.

Reply:
@bruno-garcia another way would be to provide a `removeIntegration(Class class)` type and we remove by type safely.

Reply:
That's [what I added to the .NET SDK](https://github.com/getsentry/sentry-dotnet/blob/master/src/Sentry/SentryOptionsExtensions.cs#L56-L57):

```csharp
        internal static void RemoveIntegration<TIntegration>(this SentryOptions options) where TIntegration : ISdkIntegration
            => options.Integrations = options.Integrations.RemoveAll(p => p.GetType() == typeof(TIntegration));
```

I'm not a fan since we can have more than once instance taking a strategy, for example we have here the FileObserver bits, there are two instances.

Also, it's still confusing we have the Options flags to turn things on or off and configure them. Then this odd, by type integration thingy.

Personally I'm learning more into just "add integrations at will", configure or turn them off via SentryOptions (which can be bound to configuration files for example and do not requrie code changes)